kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/listener/TeamCityTestEngineListener.kt

```diff
@@ -33,26 +31,11 @@ class TeamCityTestEngineListener(
    private var writer = TeamCityWriter(prefix, FallbackDisplayNameFormatter.default(ProjectConfiguration()))
 
    // once a spec has completed, we want to be able to check whether any given test is
-   // a container or a leaf test, and so this map contains all test that have children
-   private val children = mutableMapOf<Descriptor, MutableList<TestCase>>()
-
-   private val results = mutableMapOf<Descriptor, TestResult>()
+   // a suite or not, and so this map marks all tests that have children
+   private val parents = mutableSetOf<TestCase>()
```

Comment:
Doesn't this need to be a thread-safe type? Multiple tests in the same spec might run simultaneously if `concurrentTests > 1`

Will there be 1 TC-listener per Spec? Otherwise clearing this on specFinished sounds like it would break on any parallelism. Maybe we should maintain a map of Spec -> parents?

Reply:
That's already how it works, this listener is wrapped in a thread safe listener. There's a comment to that effect in this file.